pkg/common-controller/snapshot_metrics.go

```diff
@@ -0,0 +1,77 @@
+package common_controller
```

Comment:
I am not convinced we need to have this wrapper. OperationStatus definitely needs an implementation, however, I am not sure about the "StartMetricsOperation" and "RecordSnapshotMetrics" calls.  Those are public methods and not called in anywhere else.

Reply:
Removed these two public wrapper functions